### PR TITLE
Replace fabricate with a script that auto-regenerates test project. (not ready for merging)

### PR DESCRIPTION
**Pull request by Osmose**
In #41 I ran into issues trying to add more complex template logic due to how we generate the template directory from a working project directory. This PR switches things around so that the test project is generated from the template directory.

To test it out for the first time, run `bin/regenerate.py` and `touch cookiecutter.json`. That should generate a new test project in `test_project/sugardough`. Then open up another shell and run `fig up` inside `test_project/sugardough`, which should run the test instance of the site (it may even reuse your existing test instance, I'm not sure how but it did for me).

The setup seems to work fine even if the sugardough directory is deleted and recreated while the docker containers are still running. Of course, edits to the Dockerfile or fig config would require a rebuild, but that's okay, I think. The good news is that this will allow us to work on the templates instead of working on the test site, allowing for more complex template logic and also for values in `cookiecutter.json` that have the same value.

This is incomplete, as it's missing documentation updates, throws weird threading errors when exiting, and needs some better logging, but I wanted feedback as to whether this is a good idea before I continue.

Thoughts?

### CODE REVIEW

**Comment by Osmose** [comment]
Oh yeah, make sure to `pip install -r requirements.txt` to get watchdog and cookiecutter installed before you run `bin/regenerate.py`.

**Comment by Osmose** [comment]
@glogiotatidis I talked with @peterbe today and he's on board with this, so I'll be working on getting this PR up to snuff. Got any opinions of your own?

**Comment by glogiotatidis** [comment]
Ideally I'd like things the way they are now: build a project and then build templates automagically based on that. That feels more natural to me. But I see the value of your approach: it solves the template generation issue without funky githooks and it's more generic. Autogeneration of test-project practically gives the same experience as if you're working on the test-project itself.

:+1: go for it!

For your todo list: 
 * I faced a couple of issues with `test_project` dir missing and permissions in that directory.

**Comment by Osmose** [comment]
Updated with documentation and a slightly more polished command. I think this is ready for review and testing. :D

**Comment by peterbe** [comment]
@Osmose you did an "r?" on irc but I see the tests are failing and it's conflicting. Do you mind fixing that first?

**Comment by Osmose** [comment]
Hey man  it looked fine at the time.

(Yes I will rectify my mistakes, pardon me DOOT DOOT DOO)

**Comment by Osmose** [comment]
Travis is still spinning on this, but I believe I've fixed the issue (style errors in wsgi.py).

File: README.md [hunk 1]
```diff
@@ -40,18 +42,47 @@ Cooking sugardough
...
+1. Clone this repository
+2. Create a [virtualenv](https://virtualenv.pypa.io/en/latest/).
+3. Install development requirements using pip:
+
+   ```sh
+   $ pip install -r requirements.txt
+   ```
+4. Update the [template directory], and [cookiecutter.json] as well with new
+   variables if needed.
+5. Run the regenration script that auto-creates a test site:
```
Comment (glogiotatidis) [inline README.md]: nit: typo `regenration`

File: {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/wsgi.py [hunk 2]
```diff
@@ -17,6 +15,9 @@
 from decouple import config
 from whitenoise.django import DjangoWhiteNoise
 
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', '{{ cookiecutter.project_name }}.settings')
```
Comment (glogiotatidis) [inline {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/wsgi.py]: I think this line has to before `from django.conf import settings`. `django-admin startproject` generates `wsgi.py` this way.

File: bin/regenerate.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+def main(watch):
+    # Regenerate at least once.
+    regenerate()
+
+    if watch:
+        observer = Observer()
+
+        # Observe both the template directory and cookiecutter.json.
+        observer.schedule(RegenerateSugardoughHandler(), TEMPLATEDIR, recursive=True)
+        observer.schedule(RegenerateSugardoughHandler(patterns=['*/cookiecutter.json']), BASEDIR)
```
Comment (glogiotatidis) [inline bin/regenerate.py]: This works but that `*` is a bit confusing. I'd change it to `os.path.join(BASEDIR, 'cookiecutter.json')`

**Comment by glogiotatidis** [comment]
nice job @Osmose :+1: r+wc

**Pull request merged**